src/module/ModuleTypes.jl

```diff
@@ -41,7 +41,7 @@ This needs to be indicated due to the fact that Singulars
 vectors and polys are both stored in the poly data structure.
 """
 function (R::PolyRing{T})(m::libSingular.poly_ptr, ::Val{:vector}) where T
-    return svector{T}(R, 1, m)
+   return svector{T}(R, 1, m)
```

Comment:
I actually indented a few extra line to 3 spaces after @rfourquet, including this one, which turns out to be uncovered. To fix it, I would add a `jldoctest` example, but while thinking about one, I got some doubts why this method exists in the first place -- why would I prefer writing `R(poly, Val(:vector))` over `svector(R, poly)`? Isn't the latter easier to understand in the first place? What am I missing? @wbhart ?

Reply:
Why would these be equivalent? R here is a polynomial ring, so the result should be a polynomial. I'm not sure what this is doing.

I've got my doubts that a user should be constructing an svector directly, rather than doing a coercion into a vector space or using some more standard constructor matching what we do in AbstractAlgebra, but if this is internal then I suppose the second option looks a little better.

Reply:
@wbhart I thought this was your code, but I was not digging deep enough in the `git blame` output, this is Sebastian's code. And it was added with no test or example or justification for why it is a good idea.

My question was whether this function (which I only touched in terms of indentation) even makes sense -- your confused comment seems to confirm my feeling that it does not make sense, as it returns a vector, not a polynomial. So let's just remove it?

Reply:
Fine by me, assuming it's not used by something elsewhere in Singular.jl.

Reply:
Ahh, wait, now I remember what this was for (but aaargh, this is badly documented); It is used by the magic code in `convert_return_value` from `src/caller.jl`; the `Val` is constructed from the data that ultimately comes from the `casting_functions_pre` dictionary.